doc/specs/#1620 - Query Refinement for Disambiguation.md

```diff
@@ -0,0 +1,275 @@
+---
+author: Darshan Rander @SirusCodes and Kaleb Luedtke @Trenly
+created on: 2022-10-29
+last updated: 2022-11-28
+issue id: 1620
+---
+
+# Query Refinement for Disambiguation
+
+For [#1620](https://github.com/microsoft/winget-cli/issues/1620)
+
+## Abstract
+
+When the user runs the install command and the winget client cannot disambiguate the query down to a single package, the client will list potential matches. The user then needs to re-run the install command with the package identifier to install the package.
+
+The UX can be improved by giving users an option to select the package instead of forcing the command to be re-run with the package identifier.
+This selection option can also be extended to the `upgrade`, `show` and `uninstall` commands.
+
+## Inspiration
+
+There are many similarly-named packages in winget. When users try to install any application, there is a high chance of a collision causing multiple suggestions in the returned table. It is an annoyance to copy the package identifier and re-run the command.
+
+## Solution Design
+
+When the query for `install`, `upgrade`, `show`, or `uninstall` cannot be disambiguated to a single package, the user should be given the opportunity to continuously refine their query.
+
+> Query refinement should not occur if `--diable-interactivity` is used or if interactivity has been disabled through settings. Query refinement is not required for COM callers.
+
+### Settings
+
+This shall be an **opt-in** only feature. The setting for this feature shall be placed uder the `interactivity` object. 
+
+In order to enable the query refinement feature -
+> Note: The setting could also be called `packageDisambiguation`
+```json
+"interactivity" : {
+   "refineAmbiguousQueries": true
+}
+```
+In order to disable the query refinement feature -
+```json
+"interactivity" : {
+   "refineAmbiguousQueries": false // This is the default behavior
+}
+```
+
+From the command line, users should be able to use `--refine-query` to override the user setting and show the query refinement, and `--no-refinement` to override the user setting and not show the query refinement.
+
+### Refinement Source
+The results from the query needing refinement should be placed into a temporary database to be used for further refinement. This database should be treated as a new temporary source to be queried. Upon completion of the command, this temporary database shall be removed. If the database is already existing and a new command is run, the database shall be overwritten.
```

Comment:
I don't see any need for a file for this purpose.  Even if SQLite is used, it can be in memory.